### PR TITLE
fix condition of adding asterix, when save reader of path

### PR DESCRIPTION
**Pull request by Czaki**
# References and relevant issues

# Description
Current logic for saving a preference from the reader dialog leads to adding `*` to absolute paths, making it impossible to save a reader preference for a zarr directory. This was because an asterisk was added whenever the opened path had an extension.

This PR fixes checking if the path is an absolute path and doesn't add an asterisk if it's a directory.

### CODE REVIEW

**Review by DragaDoncila — COMMENTED** [review]
@Czaki would be good if we could add a test to cover that line- there's an existing similar [test](https://github.com/napari/napari/blob/main/napari/_qt/dialogs/_tests/test_reader_dialog.py#L185) here you should be able to edit. I also updated the description.

File: napari/_qt/dialogs/_tests/test_reader_dialog.py
Comment (Czaki) [inline napari/_qt/dialogs/_tests/test_reader_dialog.py:176]: If we depend only on QtViewer we should use only QtViewer not whole viewer.

File: napari/_qt/dialogs/_tests/test_reader_dialog.py
Comment (DragaDoncila) [inline napari/_qt/dialogs/_tests/test_reader_dialog.py]: @Czaki I'll approve once CI is green but just wanted to flag this comment needs updating. Idm if you wait until this CI run is done and I've approved though

Comment (Czaki) [reply napari/_qt/dialogs/_tests/test_reader_dialog.py]: fixed

**Comment by jni** [comment]
@Czaki @DragaDoncila I think this is ready to merge?

**Comment by Czaki** [comment]
From my side, yes.

**Review by jni — APPROVED** [review]

**Review by DragaDoncila — APPROVED** [review]
Thanks for adding the test @Czaki, lgtm!

**Pull request merged**
